src/ZODB/tests/testThreadedShutdown.py

```diff
@@ -47,8 +50,8 @@ def check_shutdown(self):
         # have different contents.
         self._db.close()
 
-    def tearDown(self):
-        ZODB.tests.util.TestCase.tearDown(self)
+        # Be sure not to 'leak' the running thread.
+        client_thread.join()
```

Comment:
I'd be tempted to add `timeout=10` or something because tests that hang are not fun at all.